core/src/test/scala/unit/kafka/admin/LeaderElectionCommandTest.scala

```diff
@@ -0,0 +1,353 @@
+      val assignment = Seq(broker2, broker3)
+
+      TestUtils.createTopic(zkClient, topic, Map(partition -> assignment), servers)
+
+      val topicPartition = new TopicPartition(topic, partition)
+
+      waitForLeaderToBecome(client, topicPartition, Option(broker2))
+
+      servers(broker3).shutdown()
+      waitForBrokerOutOfIsr(client, Set(topicPartition), broker3)
+      servers(broker2).shutdown()
+      waitForLeaderToBecome(client, topicPartition, None)
+      servers(broker3).startup()
+
+      LeaderElectionCommand.main(
+        Array(
+          "--bootstrap-server", bootstrapServers(servers),
+          "--election-type", "unclean",
+          "--topic", topic,
+          "--partition", partition.toString
+        )
+      )
+
+      assertEquals(Option(broker3), currentLeader(client, topicPartition))
+    }
+  }
+
+  @Test
+  def testPathToJsonFile(): Unit = {
+    TestUtils.resource(JAdminClient.create(createConfig(servers).asJava)) { client =>
+      val topic = "unclean-topic"
+      val partition = 0
+      val assignment = Seq(broker2, broker3)
+
+      TestUtils.createTopic(zkClient, topic, Map(partition -> assignment), servers)
+
+      val topicPartition = new TopicPartition(topic, partition)
+
+      waitForLeaderToBecome(client, topicPartition, Option(broker2))
+
+      servers(broker3).shutdown()
+      waitForBrokerOutOfIsr(client, Set(topicPartition), broker3)
+      servers(broker2).shutdown()
+      waitForLeaderToBecome(client, topicPartition, None)
+      servers(broker3).startup()
+
+      val topicPartitionPath = tempTopicPartitionFile(Set(topicPartition))
+
+      LeaderElectionCommand.main(
+        Array(
+          "--bootstrap-server", bootstrapServers(servers),
+          "--election-type", "unclean",
+          "--path-to-json-file", topicPartitionPath.toString
+        )
+      )
+
+      assertEquals(Option(broker3), currentLeader(client, topicPartition))
+    }
+  }
+
+  @Test
+  def testPreferredReplicaElection(): Unit = {
+    TestUtils.resource(JAdminClient.create(createConfig(servers).asJava)) { client =>
+      val topic = "unclean-topic"
+      val partition = 0
+      val assignment = Seq(broker2, broker3)
+
+      TestUtils.createTopic(zkClient, topic, Map(partition -> assignment), servers)
+
+      val topicPartition = new TopicPartition(topic, partition)
+
+      waitForLeaderToBecome(client, topicPartition, Option(broker2))
+
+      servers(broker2).shutdown()
+      waitForLeaderToBecome(client, topicPartition, Some(broker3))
+      servers(broker2).startup()
+      waitForBrokerInIsr(client, Set(topicPartition), broker2)
+
+      LeaderElectionCommand.main(
+        Array(
+          "--bootstrap-server", bootstrapServers(servers),
+          "--election-type", "preferred",
+          "--all-topic-partitions"
+        )
+      )
+
+      assertEquals(Option(broker2), currentLeader(client, topicPartition))
+    }
+  }
+
+  @Test
+  def testTopicWithoutPartition(): Unit = {
```

Comment:
Is it worthwhile having a `testPartitionWithoutTopic`?

Reply:
Added the test. I remove jopt-simple required options implementation as it wasn't flexible enough for our use case. I manually implemented the validation instead.